pypeit/core/scattlight.py

```diff
@@ -0,0 +1,308 @@
+""" Module for sky subtraction
+
+.. include common links, assuming primary doc root is up one directory
+.. include:: ../include/links.rst
+"""
+import numpy as np
+
+from scipy.optimize import least_squares
+from scipy import signal, interpolate, ndimage
+from IPython import embed
+
+from pypeit import msgs, utils
+
+
+def scattered_light_model(param, img):
+    """ Model used to calculate the scattered light.
+
+    The current model to generate the scattered light is a shifted, scaled, and blurred version of the
+    data recorded on the detector. We allow for a shift in the spatial and spectral direction, and the
+    blurring kernel can have different widths in the spectral and spatial directions. Finally, the model
+    includes a polynomial correction in the spectral direction to account for a change in the spectral
+    efficiency of the scattered light relative to the detector image.
+
+    Parameters
+    ----------
+    param : `numpy.ndarray`_
+        Model parameters that determine the scattered light based on the input img.
+        Here is a list of the individual parameter meanings:
+
+        * param[0] = Gaussian kernel width in the spectral direction
+        * param[1] = Gaussian kernel width in the spatial direction
+        * param[2] = Lorentzian kernel width in the spectral direction
+        * param[3] = Lorentzian kernel width in the spatial direction
+        * param[4] = Pixel shift of the scattered light in the spectral direction
+        * param[5] = Pixel shift of the scattered light in the spatial direction
+        * param[6] = Zoom factor of the scattered light (~1)
+        * param[7] = Kernel angle
+        * param[8] = Relative importance of Gaussian vs Lorentzian.
+                     0 < value < 1 means Lorentzian is weighted more
+                     value > 1 means Gaussian is weighted more.
+        * param[9:] = Polynomial scaling coefficients
+    img : `numpy.ndarray`_
+        Raw image that you want to compute the scattered light model.
+        shape is (nspec, nspat)
+        Model used to calculate the scattered light. This function is used to
+        generate a model of the scattered light, based on a set of model parameters
+        that have been optimized using self.scattered_light().
+
+    Returns
+    -------
+    model : `numpy.ndarray`_
+        Model of the scattered light for the input
+    """
+    # Extract the parameters into more conveniently named variables
```

Comment:
Is there a reason why you don't just use the simpler: 

sigmx_g, sigmy_g, sigmx_l, sigmy_l, shft_spec, shft_spat, zoom_spec, zoom_spat, constant, kern_angle, kern_scale, polyterms_spat, polyterms_spec = param